addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/handler/DeviceHandler.java

```diff
@@ -0,0 +1,208 @@
+import static org.openhab.binding.lightify.internal.LightifyConstants.RGBW_CHANNEL_ID_TEMPERATURE;
+import static org.openhab.binding.lightify.internal.LightifyConstants.SB_CHANNEL_ID_DIMMER;
+import static org.openhab.binding.lightify.internal.LightifyConstants.SB_CHANNEL_ID_POWER;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_BULB_RGBW;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_BULB_SB;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_BULB_TW;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_ZONE;
+import static org.openhab.binding.lightify.internal.LightifyConstants.TW_CHANNEL_ID_DIMMER;
+import static org.openhab.binding.lightify.internal.LightifyConstants.TW_CHANNEL_ID_POWER;
+import static org.openhab.binding.lightify.internal.LightifyConstants.TW_CHANNEL_ID_TEMPERATURE;
+
+/**
+ * <p>The {@link org.eclipse.smarthome.core.thing.binding.ThingHandler} implementation to handle commands
+ * for a OSRAM Lightify light bulb / stripe or zone (group of lights).</p>
+ * <p>Commands are sent through the Lightify gateway the device is paired to. The connection link, managed
+ * by the assigned {@link GatewayHandler} will be used to forward the command to.</p>
+ *
+ * @author Christoph Engelbert (@noctarius2k) - Initial contribution
+ */
+public class DeviceHandler extends BaseThingHandler {
+
+    /**
+     * Supported {@link ThingTypeUID}s for this handler
+     */
+    public static final Set<ThingTypeUID> SUPPORTED_TYPES = Collections.unmodifiableSet(new HashSet<ThingTypeUID>() {
+        {
+            add(THING_TYPE_LIGHTIFY_BULB_SB);
+            add(THING_TYPE_LIGHTIFY_BULB_TW);
+            add(THING_TYPE_LIGHTIFY_BULB_RGBW);
+            add(THING_TYPE_LIGHTIFY_ZONE);
+        }
+    });
+
+    private final Logger logger = LoggerFactory.getLogger(DeviceHandler.class);
+
+    private volatile ScheduledFuture<?> updateTaskRegistration;
+
+    public DeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        Runnable statusUpdater = () -> {
+            GatewayHandler gatewayHandler = getGatewayHandler();
+            LightifyLink lightifyLink = gatewayHandler.getLightifyLink();
+            LightifyLuminary luminary = getLuminary();
+            if (luminary instanceof LightifyLight) {
+                lightifyLink.performStatusUpdate(luminary, this::updateState);
+            }
+        };
+
+        updateTaskRegistration = scheduler.scheduleWithFixedDelay(statusUpdater, 1, 10, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        updateTaskRegistration.cancel(true);
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command);
```

Comment:
Please catch your IllegalStateExceptions here: I was playing around with the paperui buttons and sliders of my osram bulb and it seems that if you click rather fast, at some point the bulb is not reacting anymore. Here is one example stacktrace that occured:

2017-04-18 13:29:20.958 [ERROR] [.c.thing.internal.ThingManager:369  ] - Exception occured while calling handler: java.lang.IllegalStateException: Illegal packet type: 0x0, command: LIGHT_SWITCHjava.util.concurrent.ExecutionException: java.lang.IllegalStateException: Illegal packet type: 0x0, command: LIGHT_SWITCH
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:220)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:189)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:83)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:67)
	at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:358)
	at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:46)
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)
	at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:181)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
Caused by: java.lang.IllegalStateException: Illegal packet type: 0x0, command: LIGHT_SWITCH
	at org.openhab.binding.lightify.internal.link.LightifyLink.handleHeader(LightifyLink.java:367)
	at org.openhab.binding.lightify.internal.link.LightifyLink.onPacket(LightifyLink.java:181)
	at org.openhab.binding.lightify.internal.link.LightifyLink.readPacket(LightifyLink.java:385)
	at org.openhab.binding.lightify.internal.link.LightifyLink.perform(LightifyLink.java:157)
	at org.openhab.binding.lightify.internal.link.LightifyLink.performSwitch(LightifyLink.java:98)
	at org.openhab.binding.lightify.internal.link.LightifyLuminary.setSwitch(LightifyLuminary.java:43)
	at org.openhab.binding.lightify.handler.DeviceHandler.handlePowerSwitch(DeviceHandler.java:162)
	at org.openhab.binding.lightify.handler.DeviceHandler.handleCommand(DeviceHandler.java:107)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$4.call(ThingManager.java:361)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$4.call(ThingManager.java:1)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:218)